doc/languages-frameworks/beam.section.md

```diff
@@ -44,10 +44,24 @@ There is also a `buildMix` helper, whose behavior is closer to that of `buildErl
 
 ## How to Install BEAM Packages {#how-to-install-beam-packages}
 
-BEAM builders are not registered at the top level, simply because they are not relevant to the vast majority of Nix users. To install any of those builders into your profile, refer to them by their attribute path `beamPackages.rebar3`:
+BEAM builders are not registered at the top level, simply because they are not relevant to the vast majority of Nix users.
+To use any of those builders into your environment, refer to them by their attribute path `beamPackages.rebar3`:
+
+Example: Ephemeral shell
 
 ```ShellSession
-$ nix-env -f "<nixpkgs>" -iA beamPackages.rebar3
+$ nix-shell -p beamPackages.rebar3
+```
+
+Example: Declarative shell
+
+```nix
+let
+  pkgs = import <nixpkgs> { config = {}; overlays = []; };
+in
+pkgs.mkShell {
+  packages = with pkgs.beamPackages; [ rebar3 erlang-ls ];
```

Comment:
If you don't add a further package, you avoid showing the user two things at the same time:

```suggestion
  packages = with pkgs.beamPackages; [ rebar3 ];
```